src/main/java/software/amazon/awssdk/crt/io/ClientBootstrap.java

```diff
@@ -24,17 +24,30 @@
  * a client context for all protocol stacks in the AWS Common Runtime.
  */
 public final class ClientBootstrap extends CrtResource implements Closeable {
+    private HostResolver hostResolver;
     private EventLoopGroup elg;
 
     /**
      * Creates a new ClientBootstrap. Most applications will only ever need one instance of this.
+     * @param numThreads The number of Threads to use in the EventLoop
+     * @throws CrtRuntimeException If the system is unable to allocate space for a native client bootstrap object
+     */
+    public ClientBootstrap(int numThreads) throws CrtRuntimeException {
+        this.elg = new EventLoopGroup(numThreads);
+        this.hostResolver = new HostResolver(elg);
+    }
+
+    /**
+     * Creates a new ClientBootstrap. Most applications will only ever need one instance of this.
+     * @param hr A HostResolver instance, most applications only ever have one
      * @param elg An EventLoopGroup instance, most applications only ever have one
      * @throws CrtRuntimeException If the provided EventLoopGroup is null or invalid,
      * or if the system is unable to allocate space for a native client bootstrap object
      */
-    public ClientBootstrap(EventLoopGroup elg) throws CrtRuntimeException {
+    public ClientBootstrap(HostResolver hr, EventLoopGroup elg) throws CrtRuntimeException {
+        this.hostResolver = hr;
         this.elg = elg;
-        acquire(clientBootstrapNew(elg.native_ptr()));
+        acquire(clientBootstrapNew(elg.native_ptr(), hostResolver.native_ptr()));
     }
 
     @Override
```

Comment:
Need to close the elg and hostResolver if the ClientBootstrap owns them in close()

Reply:
Fixed